diff -Naur xwinmosaic/src/main.c xwinmosaic.0/src/main.c
--- xwinmosaic/src/main.c	1970-01-01 01:00:01.000000000 +0100
+++ xwinmosaic.0/src/main.c	2018-11-09 09:52:29.716620179 +0000
@@ -65,6 +65,7 @@
   gboolean permissive;
   gboolean persistent;
   gboolean format;
+  gboolean print_window_id;
   gboolean screenshot;
   guchar color_offset;
   gint screenshot_offset_x;
@@ -88,6 +89,8 @@
 {
   { "persistent", 'R', 0, G_OPTION_ARG_NONE, &options.persistent,
     "Make XWinMosaic acts like Alt-Tab switcher", NULL },
+  { "print-window-id" , 'w', 0, G_OPTION_ARG_NONE, &options.print_window_id,
+    "Print the X window ID rather than switching to the window", NULL },
   { "read-stdin", 'r', 0, G_OPTION_ARG_NONE, &options.read_stdin,
     "Read items from stdin (and print selected item to stdout)", NULL },
   { "permissive", 'p', 0, G_OPTION_ARG_NONE, &options.permissive,
@@ -431,16 +434,17 @@
 static void on_rect_click (GtkWidget *widget, gpointer data)
 {
   MosaicWindowBox *box = MOSAIC_WINDOW_BOX (widget);
-  Window win_to_switch;
-  if (!options.read_stdin) {
-    win_to_switch = mosaic_window_box_get_xwindow (box);
+  
+
+  if (options.read_stdin) {
+    puts(mosaic_window_box_get_name (box));
+  } else if (options.print_window_id) {
+    printf("%ld\n", mosaic_window_box_get_xwindow(box));
   } else {
-    puts (mosaic_window_box_get_name (box));
-  }
-  gtk_widget_hide (window);
-  if (!options.read_stdin) {
-    switch_to_window (win_to_switch);
+    gtk_widget_hide (window);
+    switch_to_window(mosaic_window_box_get_xwindow(box));
   }
+  
   if (!options.persistent) {
     gtk_main_quit ();
   }
